portfolio/src/main/java/com/google/sps/validators/NameValidator.java

```diff
@@ -0,0 +1,22 @@
+package com.google.sps.validators;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class NameValidator implements TextValidator {
```

Comment:
I think you could get away with only a single class, and no interfaces instead of two classes and one interface. The current `NameValidator` and the `CommentValidator` are almost the same, except for a different value for the constant; also if you added another field, let's say "nickname" or "subject", would you have to add two more classes?

Instead, you could just write a single `class TextValidator` with a single method `boolean validate(String value)` and constructor parameters that configure the validator (e.g. for `lengthLimit` and `validPattern`).